src/apollo/client.ts

```diff
@@ -18,6 +19,22 @@ const authLink = setContext((_, { headers }) => {
   };
 });
 
+const typePolicies: TypedTypePolicies = {
+  UserDetails: {
+    fields: {
+      authenticated: {
+        read(value = false): boolean {
+          return value;
+        },
```

Comment:
Instead of handling it manually, you should return the value based on existing cache configuration. I believe the `readField` function should work here or you could read the `cache` object directly (more info: https://www.apollographql.com/docs/react/caching/cache-field-behavior/#fieldpolicy-api-reference)
```suggestion
        read(_, { readField }): boolean {
          return !!readField("id");
        },
```